wifiphisher/common/interfaces.py

```diff
@@ -145,15 +146,65 @@ def __init__(self, name):
         self._support_ap_mode = False
         self._support_monitor_mode = False
         self.being_used = False
+        self._prev_mac = None
+        self._current_mac = None
 
         # Set monitor and AP mode if card supports it
         card = pyw.getcard(name)
         modes = pyw.devmodes(card)
+        mac = pyw.macget(card)
 
         if "monitor" in modes:
             self._support_monitor_mode = True
         if "AP" in modes:
             self._support_ap_mode = True
+        #set the current and prev mac to the origianl mac
+        self._prev_mac = mac
+        self._current_mac = mac
+
+    def _generate_random_address(self):
+        """
+        Make and return the randomized MAC address
+
+        :param self: A NetworkAdapter object
+        :type self: NetworkAdapter
+        :return: A MAC address
+        :rtype str
+        """
+
+        mac_addr = DEFAULT_OUI + ":{:02x}:{:02x}:{:02x}".format(random.randint(0, 255),\
+                random.randint(0, 255), random.randint(0, 255))
+        return mac_addr
+
+    def randomize_interface_mac(self, mac=None):
+        """
+        Randomize the MACs for the network adapters
+
+        :param self: A NetworkAdapter object
+        :param mac: A MAC address 
+        :type self: NetworkAdapter
+        :type mac: string
+        :return: None
+        :rtype: None
+        """
+        if mac == None:
+            mac_addr = self._generate_random_address()
+        else:
+            mac_addr = mac
+
+        card = pyw.getcard(self.get_name())
+        pyw.down(card)
+
+        try:
+            pyw.macset(card, mac_addr)
+        except pyric.error:
+            print ("\n[" + R + "!" + W + "] Specified MAC Address is " 
+                    + 'invalid for ' + self.get_name() +
+                    " generate random mac address")
+            mac_addr = self._generate_random_address()
+            pyw.macset(card, mac_addr)
```

Comment:
Actually I think it will not fail, since the only fail point is for the mis-using of OUI. I can still raise exception and catch it in `pywifiphisher.py`